mpas_analysis/shared/io/namelist_streams_interface.py

```diff
@@ -144,17 +146,20 @@ def readpath(self, streamName, startDate=None, endDate=None):
         on or before the endDate are included in the file list.
         """
         template = self.read(streamName, 'filename_template')
-        replacements = {'$Y':'[0-9][0-9][0-9][0-9]',
-                        '$M':'[0-9][0-9]',
-                        '$D':'[0-9][0-9]',
-                        '$S':'[0-9][0-9][0-9][0-9][0-9]',
-                        '$h':'[0-9][0-9]',
-                        '$m':'[0-9][0-9]',
-                        '$s':'[0-9][0-9]'}
+        if template is None:
+            raise ValueError('Stream {} not found in streams file {}.'.format(
+                streamName, self.fname))
```

Comment:
I agree.  We can hopefully use the testing framework to ensure error checking works correctly too, which makes verification a lot easier.  I think we are doing a good job moving in this direction!